symfony/src/Controller/PatientStatusController.php

```diff
@@ -0,0 +1,180 @@
+            'imports' => $patientStatusImports
+        ]);
+    }
+
+    /**
+     * @Route("/patient/status/confirmation/{id}", name="patientStatusImportConfirmation", methods={"GET", "POST"})
+     */
+    public function patientStatusImportConfirmation(int $id, Request $request, EntityManagerInterface $em, LoggerService $loggerService)
+    {
+        $patientStatusImport = $em->getRepository(PatientStatusImport::class)->findOneBy(['id' => $id, 'userId' => $this->getUser()->getId(), 'confirm' => 0]);
+        if (empty($patientStatusImport)) {
+            throw $this->createNotFoundException('Page Not Found!');
+        }
+        $form = $this->createForm(PatientStatusImportConfirmFormType::class);
+        $form->handleRequest($request);
+        $importPatientStatuses = $patientStatusImport->getPatientStatusTemps()->slice(0, 100);
+        if ($form->isSubmitted() && $form->isValid()) {
+            if ($form->get('Confirm')->isClicked()) {
+                $batchSize = 50;
+                foreach ($importPatientStatuses as $key => $importPatientStatus) {
+                    $patientStatus = $em->getRepository(PatientStatus::class)->findOneBy([
+                        'participantId' => $importPatientStatus->getParticipantId(),
+                        'organization' => $patientStatusImport->getOrganization()->getName()
+                    ]);
+                    if (!$patientStatus) {
+                        $patientStatus = new PatientStatus();
+                        $patientStatus
+                            ->setParticipantId($importPatientStatus->getParticipantId())
+                            ->setAwardee($patientStatusImport->getAwardee())
+                            ->setOrganization($patientStatusImport->getOrganization()->getName());
+                        $em->persist($patientStatus);
+                    }
+                    $patientStatusHistory = new PatientStatusHistory();
+                    $patientStatusHistory
+                        ->setUserId($patientStatusImport->getUserId())
+                        ->setSite($patientStatusImport->getSite())
+                        ->setStatus($importPatientStatus->getStatus())
+                        ->setComments($importPatientStatus->getComments())
+                        ->setCreatedTs(new \DateTime())
+                        ->setPatientStatus($patientStatus)
+                        ->setImport($patientStatusImport);
+                    $em->persist($patientStatusHistory);
+
+                    // Update history id in patient_status table
+                    $patientStatus->setHistory($patientStatusHistory);
+                    if (($key % $batchSize) === 0) {
+                        $em->flush();
+                        $loggerService->log(Log::PATIENT_STATUS_HISTORY_ADD, $patientStatusHistory->getId());
+                        $loggerService->log(Log::PATIENT_STATUS_EDIT, $patientStatus->getId());
```

Comment:
Do we need these log entries? We are already logging the patient status import entry, and each history item is connected to the import entry. Also, the way it is written here, it will only log the last status and status history action in each batch.

Another factor is that we haven't implemented the buffered logger yet, so this would generate a bunch of remote calls to Stackdriver.